python/src/aiconfig/editor/client/src/components/GlobalParametersContainer.tsx

```diff
@@ -1,21 +1,43 @@
-import { Accordion, Text } from "@mantine/core";
+import { Accordion, Text, createStyles } from "@mantine/core";
 import { JSONObject } from "aiconfig";
-import { memo, useState } from "react";
+import { memo, useContext, useState } from "react";
 import ParametersRenderer from "./ParametersRenderer";
+import AIConfigContext from "../contexts/AIConfigContext";
 
 type Props = {
   initialValue: JSONObject;
   onUpdateParameters: (newParameters: JSONObject) => void;
 };
 
+const useStyles = createStyles(() => ({
+  parametersContainer: {
+    margin: "16px auto 16px 36px",
+  },
+  parametersContainerReadonly: {
+    margin: "16px auto",
+  },
+}));
+
 export default memo(function GlobalParametersContainer({
   initialValue,
   onUpdateParameters,
 }: Props) {
   const [isParametersDrawerOpen, setIsParametersDrawerOpen] = useState(false);
 
+  const { classes } = useStyles();
+  const { readOnly } = useContext(AIConfigContext);
+
   return (
-    <div className="parametersContainer">
+    // Set local and global classname. Global will override if specified
+    // Local is readonly or not. Global will always have parametersContainer class
+    // and if readonly, will also have parametersContainerReadonly class (to allow overrides)
+    <div
+      className={`${
+        readOnly
+          ? classes.parametersContainerReadonly
+          : classes.parametersContainer
+      } parametersContainer ${readOnly ? "parametersContainerReadonly" : ""}`}
```

Comment:
K yo newbie mcscrewbie over here: what do the 2nd and 3rd args do for the `className` param: https://stackoverflow.com/questions/23663151/whats-the-point-of-the-second-argument-in-element-classlist-toggle

Reply:
Classname only takes in a string. When we use `useStyles`, the classes we define are actually serialized to class name and values used for the element styles. It's a bit of a trick, but we can actually apply multiple class names by just defining more as part of the string literal. e.g. `className="classA classB classC" would apply all 3 classes to the element.
In our case, we always want the literal `parametersContainer` class applied (so we can override it in the theme styles). We also want our local styles (from `classes.parametersContainer` or `parametersContainerReadonly`) applied. And lastly, we want the literal `parametersContainerReadonly` applied if readonly so that we can use it in the global theme styles (where 'readOnly' state is not always available)

Reply:
Ahhhhh I understand thanks for explaining. I am familiar with his from React `styles={[]}` but didn't realize `classname` also did that, cool!